src/server/jenkinsApi.js

```diff
@@ -2,130 +2,166 @@ const express = require('express');
 const router = express.Router();
 const axios = require('axios');
 
-const QA_BASE_URL = 'http://ci.qfun.com:8080/job/pureconnect/job/interaction_connect/job/connect_cic_regression';
-const MAIN_BASE_URL = 'http://ci.qfun.com:8080/job/pureconnect/job/interaction_connect/job/connect_main_regression';
-const JSON_API = '/api/json';
-const LAST_SUCCESSFUL_ROUTE = '/lastSuccessfulBuild/testReport';
-
-const QA_JOB_URL_API = QA_BASE_URL + JSON_API;
-const QA_LAST_SUCCESSFUL_TEST_REPORT_URL = QA_BASE_URL + LAST_SUCCESSFUL_ROUTE;
-const QA_LAST_SUCCESSFUL_TEST_REPORT_URL_API = QA_BASE_URL + LAST_SUCCESSFUL_ROUTE + JSON_API;
+const WEBIC_ICAT_URL = 'http://ci.qfun.com:8080/job/pureconnect/job/interaction_connect/job/webic-icat/api/json'
+const MAIN_BASE_URL = 'http://ci.qfun.com:8080/job/pureconnect/job/interaction_connect/job/webic-icat/job/client.test.latest_systest/';
+const JSON_API = 'api/json';
+const LAST_SUCCESSFUL_ROUTE = 'lastSuccessfulBuild/testReport/';
+
 const MAIN_JOB_URL_API = MAIN_BASE_URL + JSON_API;
 const MAIN_LAST_SUCCESSFUL_TEST_REPORT_URL = MAIN_BASE_URL + LAST_SUCCESSFUL_ROUTE;
 const MAIN_LAST_SUCCESSFUL_TEST_REPORT_URL_API = MAIN_BASE_URL + LAST_SUCCESSFUL_ROUTE + JSON_API;
 
+let QA_BASE_URL = 'http://ci.qfun.com:8080/job/pureconnect/job/interaction_connect/job/connect_cic_regression/';
+let QA_JOB_URL_API = QA_BASE_URL + JSON_API;
+let QA_LAST_SUCCESSFUL_TEST_REPORT_URL = QA_BASE_URL + LAST_SUCCESSFUL_ROUTE;
+let QA_LAST_SUCCESSFUL_TEST_REPORT_URL_API = QA_BASE_URL + LAST_SUCCESSFUL_ROUTE + JSON_API;
+
+let jobList = [];
+
+let getIcatJobs = () => {
+    return axios.get(WEBIC_ICAT_URL).then((result) => {
+        result.data.jobs.forEach( (job) => {
+            jobList.push(job);
+        });
+    });
+};
+
+let getQAJobUrl = () => {
```

Comment:
We should be able to declare this as a function too.

I also think it would be better if this just returned the new QA URL rather than modify the state of a shared variable.